### PR TITLE
Task/ldap helper

### PR DESCRIPTION
**Pull request by AlvaroVega**
- [x] Implement new API
- [x] Apiary doc
- [x] Doc about new endponts
- [x] Docker ENV update
- [x] Tests

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -1,6 +1,9 @@
 CHANGES
 =======
 
+FEATURE: Ldap support for user management
+
+1.7.0
```
Comment (fgalan) [inline ChangeLog]: I understand that the plan is to close release 1.7.0 before adding the LDAP support.

If that is correct, I'd sugges to do close the 1.7.0 release in an independent PR (mirroring this one for instance PR https://github.com/telefonicaid/orchestrator/pull/100)

Comment (fgalan) [reply ChangeLog]: Version1.7.0 closed in PR https://github.com/telefonicaid/orchestrator/pull/118

NTC

File: CONFIG.md [hunk 2]
```diff
@@ -18,6 +18,8 @@ PEP_PERSEO = {}
 STH = {}
 PERSEO = {}
 CYGNUS = {}
+LDAP = {}
```
Comment (fgalan) [inline CONFIG.md]: Maybe an example could be included below, as with other endpoints.

File: ChangeLog [hunk 3]
```diff
@@ -1,6 +1,9 @@
 CHANGES
 =======
 
+FEATURE: Ldap support for user management
```
Comment (fgalan) [inline ChangeLog]: Typo?

Ldap -> LDAP

File: Dockerfile [hunk 4]
```diff
@@ -3,7 +3,7 @@ FROM centos:7
 MAINTAINER Alvaro Vega <alvaro.vegagarcia@telefonica.com>
 
 ENV ORCHESTRATOR_USER orchestrator
-ENV ORCHESTRATOR_VERSION 1.6.0
+ENV ORCHESTRATOR_VERSION 1.7.0
```
Comment (fgalan) [inline Dockerfile]: ORCHESTRATOR_VERSION should be also adjusted in the release/1.7.0 branch (and the tag 1.7.0 moved to the new HEAD of release/1.7.0)?

File: apiary.apib [hunk 5]
```diff
@@ -1057,3 +1057,125 @@ The following metrics are collected by the component:
...
+                      }
+      }
++ Response 200 (application/json)
+    + Body
+      {
+         "details": "OK"
+      }
+
+
+## Ldap Auth [/v1.0/ldap/auth]
```
Comment (fgalan) [inline apiary.apib]: Typo?

Ldap - > LDAP

File: setup.cfg [hunk 6]
```diff
@@ -1,6 +1,6 @@
 [metadata]
 name = iotp_orchestrator
-version = 1.6.0
+version = 1.7.0
```
Comment (fgalan) [inline setup.cfg]: This version variable should be also adjusted in the release/1.7.0 branch (and the tag 1.7.0 moved to the new HEAD of release/1.7.0)?

File: sonar-project.properties [hunk 7]
```diff
@@ -2,7 +2,7 @@
 ############################ COMMON PROPERTIES
 sonar.projectName=Orchestrator
 sonar.projectKey=com.telefonica.iot:orchestrator
-sonar.projectVersion=1.6.0
+sonar.projectVersion=1.7.0
```
Comment (fgalan) [inline sonar-project.properties]: Same here... although, who do cares nowadays about sonar? :)

File: src/orchestrator/core/openldap.py [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+    def getUserDetail(self,
+                    USER_NAME,
+                    USER_PASSWORD):
+        try:
+            conn = self.bindUser(USER_NAME, USER_PASSWORD)
+            baseDN = "uid=" + USER_NAME + ",ou=users," + self.LDAP_BASEDN
+            searchScope = ldap.SCOPE_SUBTREE
+            ## retrieve all attributes
+            retrieveAttributes = None
+            #searchFilter = "cn=*" + USER_NAME + "*"
```
Comment (fgalan) [inline src/orchestrator/core/openldap.py]: Leftover comment?

File: Dockerfile (hunk 4 above)
Comment (AlvaroVega) [reply Dockerfile]: NTC

File: CONFIG.md (hunk 2 above)
Comment (AlvaroVega) [reply CONFIG.md]: described below

**Pull request merged**